lib/collection/src/collection.rs

```diff
@@ -1206,16 +1216,20 @@ impl Collection {
             let mut config = self.collection_config.write().await;
             config.optimizer_config = optimizer_config;
         }
-        {
-            let shard_holder = self.shards_holder.read().await;
-            for replica_set in shard_holder.all_shards() {
-                replica_set.on_optimizer_config_update().await?;
-            }
-        }
+        self.trigger_optimizers().await?;
         self.collection_config.read().await.save(&self.path)?;
         Ok(())
     }
 
+    /// Trigger the optimizers on all shards for this collection
+    async fn trigger_optimizers(&self) -> CollectionResult<()> {
+        let shard_holder = self.shards_holder.read().await;
+        for replica_set in shard_holder.all_shards() {
+            replica_set.on_optimizer_config_update().await?;
+        }
+        Ok(())
+    }
```

Comment:
Potential problem: this is blocking right now.

We might just accept that a call to update collection parameters is blocking. Seems fair enough to me. Or should changing HSNW parameters go through instantly? What do we think?

Reply:
I believe the wait time is directly proportional to the length of the update queue.
I wonder how long it could be realistically.

Tiny optimization: Can we wait on the shard's config update in parallel instead of sequentially?

Reply:
This is only blocking on existing optimizers that are being stopped, not on full HNSW regenereation for example. This is okay for now.

> Tiny optimization: Can we wait on the shard's config update in parallel instead of sequentially?

I'll do a `join_all` on it in a new commit. Thanks!